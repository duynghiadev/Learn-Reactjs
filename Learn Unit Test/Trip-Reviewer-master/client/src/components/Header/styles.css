.header-container {
  height: 55px;
  line-height: 55px;
  background: #00bcd4;
  position: relative;
  border-bottom: 6px solid #0495a8;
}

.header-container .logo {
  text-align: center;
  font-size: 30px;
  font-family: Breathe;
  letter-spacing: 3px;
  color: #ffffff;
  display: block;
}

.header-container .open-nav {
  position: absolute;
  left: 0px;
}

.header-container .only-mobile {
  display: none;
}

.header-container .only-desktop {
  position: absolute;
  color: #ffffff;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 100%;
}

@media (max-width: 800px) {
  .header-container .only-mobile {
    display: block;
  }

  .header-container .only-desktop {
    display: none;
  }
}
