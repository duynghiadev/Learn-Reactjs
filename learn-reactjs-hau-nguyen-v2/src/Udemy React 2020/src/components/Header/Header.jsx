import PropTypes from 'prop-types'
import './style.scss'

Header.propTypes = {}

const Header = () => {
  return (
    <div>
      <h1>Header</h1>
    </div>
  )
}

export default Header
