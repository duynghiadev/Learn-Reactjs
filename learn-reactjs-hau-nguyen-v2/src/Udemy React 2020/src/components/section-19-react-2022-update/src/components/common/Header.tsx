export interface HeaderProps {}

export function Header(props: HeaderProps) {
  return (
    <div>
      <h2>Header</h2>
    </div>
  )
}
