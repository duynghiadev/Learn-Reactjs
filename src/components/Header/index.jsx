import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';

function Header(props) {
  return <div>Header</div>;
}

Header.propTypes = {};

export default Header;
