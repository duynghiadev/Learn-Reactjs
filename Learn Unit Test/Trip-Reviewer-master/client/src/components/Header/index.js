import React, { Component } from 'react';
import FontAwesome from 'react-fontawesome';
import { Link } from 'react-router-dom';

import NavigationItem from '../../containers/NavigationItem';
import SideNav from '../Sidenav';
import {
  home,
  userProfile,
  signup,
  login,
  userReviews,
  addTrip,
  logout,
} from '../../constants/navigation-items';

import './styles.css';

class Header extends Component {
  state = {
    showNav: false,
  };

  handleHideNav = () => {
    window.removeEventListener('resize', this.handleHideNav);
    this.setState({ showNav: false });
  };

  handleOpenNav = () => {
    window.addEventListener('resize', this.handleHideNav);
    this.setState({ showNav: true });
  };

  render() {
    return (
      <header className="header-container">
        <div className="only-mobile">
          <div className="open-nav">
            <FontAwesome
              name="bars"
              onClick={this.handleOpenNav}
              style={{
                color: '#ffffff',
                padding: '10px',
                cursor: 'pointer',
              }}
            />
          </div>
          <SideNav showNav={this.state.showNav} onHideNav={this.handleHideNav} />
        </div>

        <div className="only-desktop">
          <div className="left-side">
            <NavigationItem item={home} />
            <NavigationItem item={userReviews} />
            <NavigationItem item={addTrip} />
          </div>

          <div className="right-side">
            <NavigationItem item={userProfile} />
            <NavigationItem item={login} />
            <NavigationItem item={signup} />
            <NavigationItem item={logout} />
          </div>
        </div>

        <Link to="/" className="logo">
          Travel Stories
        </Link>
      </header>
    );
  }
}

export default Header;
