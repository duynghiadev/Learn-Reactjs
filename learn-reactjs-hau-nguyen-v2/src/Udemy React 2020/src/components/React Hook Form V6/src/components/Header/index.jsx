import { Box, IconButton, Menu, MenuItem } from '@material-ui/core';
import AppBar from '@material-ui/core/AppBar';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import { makeStyles } from '@material-ui/core/styles';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { AccountCircle, Close } from '@material-ui/icons';
import CodeIcon from '@material-ui/icons/Code';
import Login from '../../features/Auth/components/Login';
import Register from '../../features/Auth/components/Register';
import { logout } from '../../features/Auth/userSlice';
import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link, NavLink } from 'react-router-dom';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
  link: {
    color: '#fff',
    textDecoration: 'none',
  },
  closeBtn: {
    position: 'absolute',
    top: theme.spacing(1),
    right: theme.spacing(1),
    zIndex: 1,
  },
}));

const MODE = {
  REGISTER: 'register',
  LOGIN: 'login',
};

export default function Header() {
  const dispatch = useDispatch();
  const loggedInUser = useSelector((state) => state.user.current);
  const isLoggedIn = !!loggedInUser.id;
  const classes = useStyles();
  const [open, setOpen] = useState(false);
  const [mode, setMode] = useState(MODE.LOGIN);
  const [anchorEl, setAnchorEl] = useState(null);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleClickUser = (e) => {
    setAnchorEl(e.currentTarget);
  };

  const handleCloseMenu = () => {
    setAnchorEl(null);
  };

  const handleLogoutClick = () => {
    const action = logout();
    dispatch(action);
  };

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <CodeIcon className={classes.menuButton}></CodeIcon>
          <Typography variant="h6" className={classes.title}>
            <Link className={classes.link} to="/">
              EZ Shop
            </Link>
          </Typography>
          <NavLink className={classes.link} to="/todos">
            <Button color="inherit">Todos</Button>
          </NavLink>
          <NavLink className={classes.link} to="/albums">
            <Button color="inherit">Albums</Button>
          </NavLink>

          {!isLoggedIn && (
            <Button color="inherit" onClick={handleClickOpen}>
              Login
            </Button>
          )}

          {isLoggedIn && (
            <IconButton color="inherit" onClick={handleClickUser}>
              <AccountCircle />
            </IconButton>
          )}
        </Toolbar>
      </AppBar>
      <Dialog
        disableBackdropClick
        disableEscapeKeyDown
        open={open}
        onClose={handleClose}
        aria-labelledby="form-dialog-title"
      >
        <IconButton>
          <Close className={classes.closeBtn} onClick={handleClose}></Close>
        </IconButton>
        <DialogContent>
          {mode === MODE.REGISTER && (
            <>
              <Register closeDialog={handleClose}></Register>
              <Box textAlign="center">
                <Button color="primary" onClick={() => setMode(MODE.LOGIN)}>
                  Already an account. Login here
                </Button>
              </Box>
            </>
          )}
          {mode === MODE.LOGIN && (
            <>
              <Login closeDialog={handleClose}></Login>
              <Box textAlign="center">
                <Button color="primary" onClick={() => setMode(MODE.REGISTER)}>
                  Don't have an account. Register here
                </Button>
              </Box>
            </>
          )}
        </DialogContent>
      </Dialog>
      <Menu
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleCloseMenu}
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'right',
        }}
        transformOrigin={{
          vertical: 'top',
          horizontal: 'right',
        }}
        getContentAnchorEl={null}
      >
        <MenuItem onClick={handleCloseMenu}>My account</MenuItem>
        <MenuItem onClick={handleLogoutClick}>Logout</MenuItem>
      </Menu>
    </div>
  );
}
