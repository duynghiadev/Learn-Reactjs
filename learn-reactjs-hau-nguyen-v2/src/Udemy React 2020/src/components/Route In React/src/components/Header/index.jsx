import AppBar from '@material-ui/core/AppBar'
import Button from '@material-ui/core/Button'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import { makeStyles } from '@material-ui/core/styles'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import CodeIcon from '@material-ui/icons/Code'
import { useState } from 'react'
import { Link, NavLink } from 'react-router-dom'

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1
  },
  menuButton: {
    marginRight: theme.spacing(2)
  },
  title: {
    flexGrow: 1
  },
  link: {
    color: '#fff',
    textDecoration: 'none'
  }
}))

export default function Header() {
  const [open, setOpen] = useState(false)

  const handleClickOpen = () => {
    setOpen(true)
  }

  const handleClose = () => {
    setOpen(false)
  }

  const classes = useStyles()

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <CodeIcon className={classes.menuButton} />

          <Typography variant="h6" className={classes.title}>
            <Link className={classes.link} to="/">Duy Nghia Dev</Link>
          </Typography>

          <NavLink className={classes.link} to="/todos">
            <Button color="inherit">Todos</Button>
          </NavLink>

          <NavLink className={classes.link} to="/albums">
            <Button color="inherit">Albums</Button>
          </NavLink>

          <Button color="inherit" onClick={handleClickOpen}>Register</Button>
        </Toolbar>
      </AppBar>

      <Dialog
        disableBackdropClick
        disableEscapeKeyDown
        open={open}
        onClose={handleClose}
        aria-labelledby="form-dialog-title"
      >
        <DialogContent>
          <DialogContentText>
            To subscribe to this website, please enter your email address here. We will send updates
            occasionally.
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose} color="primary">
            Cancel
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  )
}
