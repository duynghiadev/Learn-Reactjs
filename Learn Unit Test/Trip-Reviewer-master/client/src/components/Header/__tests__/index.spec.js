import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';

import Header from '..';

jest.mock('../../Sidenav', () => 'SideNav');
jest.mock('../../../containers/NavigationItem', () => 'NavigationItem');
jest.mock('react-router-dom', () => ({ Link: 'Link' }));

const mockComponent = props => <Header {...props} />;

describe('<Header />', () => {
  it('should render component', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should remove listener and change showNav state to false when handleHideNav is triggered', () => {
    const removeEventListener = jest.fn();
    const instance = mount(mockComponent()).instance();

    window.removeEventListener = removeEventListener;
    instance.handleHideNav();

    expect(instance.state.showNav).toEqual(false);
    expect(removeEventListener).toHaveBeenCalled();
  });

  it('should add listener and change showNav state to true when handleOpenNav is triggered', () => {
    const addEventListener = jest.fn();
    const instance = mount(mockComponent()).instance();

    window.addEventListener = addEventListener;
    instance.handleOpenNav();

    expect(instance.state.showNav).toEqual(true);
    expect(addEventListener).toHaveBeenCalled();
  });
});
