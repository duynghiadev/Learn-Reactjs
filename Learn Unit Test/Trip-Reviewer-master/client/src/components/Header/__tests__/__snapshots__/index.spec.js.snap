// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render component 1`] = `
<header
  className="header-container"
>
  <div
    className="only-mobile"
  >
    <div
      className="open-nav"
    >
      <span
        aria-hidden={true}
        className="fa fa-bars"
        onClick={[Function]}
        style={
          Object {
            "color": "#ffffff",
            "cursor": "pointer",
            "padding": "10px",
          }
        }
      />
    </div>
    <SideNav
      onHideNav={[Function]}
      showNav={false}
    />
  </div>
  <div
    className="only-desktop"
  >
    <div
      className="left-side"
    >
      <NavigationItem
        item="HOME"
      />
      <NavigationItem
        item="USER_REVIEWS"
      />
      <NavigationItem
        item="ADD_TRIP"
      />
    </div>
    <div
      className="right-side"
    >
      <NavigationItem
        item="USER_PROFILE"
      />
      <NavigationItem
        item="LOGIN"
      />
      <NavigationItem
        item="SIGNUP"
      />
      <NavigationItem
        item="LOGOUT"
      />
    </div>
  </div>
  <Link
    className="logo"
    to="/"
  >
    Travel Stories
  </Link>
</header>
`;
